src/main/java/org/opensearch/ad/transport/ADJobRunnerTransportAction.java

```diff
@@ -0,0 +1,120 @@
+        this.client = client;
+    }
+
+    @Override
+    protected void doExecute(Task task, ExtensionActionRequest request, ActionListener<ExtensionActionResponse> actionListener) {
+        String errorMessage = "Failed to run the Job";
+        ActionListener<ExtensionActionResponse> listener = wrapRestActionListener(actionListener, errorMessage);
+        JobRunnerRequest jobRunnerRequest = null;
+        try {
+            jobRunnerRequest = new JobRunnerRequest(request.getRequestBytes());
+        } catch (Exception e) {
+            LOG.error(e);
+            listener.onFailure(e);
+        }
+
+        try {
+            CompletableFuture<AnomalyDetectorJob> inProgressFuture = new CompletableFuture<>();
+            String jobParameterDocumentId = jobRunnerRequest.getJobParameterDocumentId();
+            if (jobParameterDocumentId == null || jobParameterDocumentId.isEmpty()) {
+                listener.onFailure(new IllegalArgumentException("jobParameterDocumentId cannot be empty or null"));
+            } else {
+                findById(jobParameterDocumentId, new ActionListener<>() {
+                    @Override
+                    public void onResponse(AnomalyDetectorJob anomalyDetectorJob) {
+                        scheduledJobParameter = anomalyDetectorJob;
+                        inProgressFuture.complete(scheduledJobParameter);
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        logger.info("could not find AnomalyDetectorJob with id " + jobParameterDocumentId, e);
+                        inProgressFuture.completeExceptionally(e);
+                    }
+                });
+                JobExecutionContext jobExecutionContext = jobRunnerRequest.getJobExecutionContext();
+
+                AnomalyDetectorJobRunner.getJobRunnerInstance().runJob(scheduledJobParameter, jobExecutionContext);
+                JobRunnerResponse jobRunnerResponse = new JobRunnerResponse(true);
+                listener.onResponse(new ExtensionJobActionResponse<>(jobRunnerResponse));
+            }
+        } catch (Exception e) {
+            LOG.error(e);
+            listener.onFailure(e);
+        }
+
+    }
+
+    private void findById(String jobParameterId, ActionListener<AnomalyDetectorJob> listener) {
+        GetRequest getRequest = new GetRequest(AnomalyDetectorJob.ANOMALY_DETECTOR_JOB_INDEX, jobParameterId);
+
+        try {
+            client.get(getRequest, ActionListener.wrap(response -> {
+                if (!response.isExists()) {
+                    listener.onResponse(null);
+                } else {
+                    try {
+                        XContentParser parser = XContentType.JSON
+                            .xContent()
+                            .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, response.getSourceAsString());
```

Comment:
Instead of passing EMPTY `NamedXContentRegistry` here. Can we pass it from the ExtensionsRunner?

`this.xContentRegistry = runner.getNamedXContentRegistry().getRegistry()`

Reply:
ACK